app/src/main/java/io/appium/settings/RecorderService.java

```diff
@@ -0,0 +1,143 @@
+    private static final String BASE = "io.appium.settings.recording";
+    public static final String ACTION_RECORDING_START = BASE + ".ACTION_START";
+    public static final String ACTION_RECORDING_STOP = BASE + ".ACTION_STOP";
+    public static final String ACTION_RECORDING_RESULT_CODE = "result_code";
+    public static final String ACTION_RECORDING_FILENAME = "recording_filename";
+
+    private static RecorderThread recorderThread;
+
+    public RecorderService() {
+        super();
+    }
+
+    @Override
+    public void onDestroy() {
+        Log.v(TAG, "onDestroy:");
+        if (recorderThread != null && recorderThread.isRecordingContinue()) {
+            recorderThread.stopRecording();
+        }
+        super.onDestroy();
+    }
+
+    @Nullable
+    @Override
+    public IBinder onBind(final Intent intent) {
+        return null;
+    }
+
+    @SuppressLint("NewApi")
+    @Override
+    public int onStartCommand(final Intent intent, final int flags, final int startId) {
+        Log.v(TAG, "onStartCommand:intent=" + intent);
+
+        int result = START_STICKY;
+        final String action = intent != null ? intent.getAction() : null;
+        Log.v(TAG, "onStartCommand:intent=" + "action string: " + action);
+        if (ACTION_RECORDING_START.equals(action)) {
+            showNotification(TAG); // TODO is this really necessary
+
+            MediaProjectionManager mMediaProjectionManager =
+                    (MediaProjectionManager) getSystemService(Context.MEDIA_PROJECTION_SERVICE);
+
+            if (mMediaProjectionManager != null) {
+                startScreenRecord(mMediaProjectionManager, intent);
+            }
+        } else if (ACTION_RECORDING_STOP.equals(action)) {
+            Log.v(TAG, "onStartCommand:intent=" + "stop");
+            stopScreenRecord();
+            result = START_NOT_STICKY;
+        }
+
+        return result;
+    }
+
+    /**
+     * start screen recording
+     */
+    @SuppressLint("NewApi")
+    private void startScreenRecord(MediaProjectionManager mediaProjectionManager,
+                                   final Intent intent) {
+        if (recorderThread == null) {
+            Log.v(TAG, "recorder thread null");
+
+            final int resultCode = intent.getIntExtra(ACTION_RECORDING_RESULT_CODE, 0);
+            // get MediaProjection
+            final MediaProjection projection = mediaProjectionManager.getMediaProjection(resultCode,
+                    intent);
+            if (projection != null) {
+                Log.v(TAG, "projection different from null");
+                final DisplayMetrics metrics = getResources().getDisplayMetrics();
+                int width = metrics.widthPixels;
+                int height = metrics.heightPixels;
+                float scale;
+                final float scaleX;
+                final float scaleY;
+                if (width > height) {
+                    // Horizontal
+                    scaleX = width / 1920f;
```

Comment:
I'm still not quite convinced about these hardcoded values. Not all phones have the same width and height. Perhaps we could read these values from the display manager instead

Reply:
+1000 yeah that was the solution what i was thinking while updating, i will look into it to make it dynamic 👍

Reply:
okay done, https://github.com/sirmordred/io.appium.settings/commit/031b55b7a4995dd5a8aaa5c1de3d1089c4b07b13 the problem was that Media Encoder supports different range of width/heights on different phones, so i clamped device specific width/height in supported range to make it dynamic 👍 also i fixed the orientation problem (it was crashing before) it now records in both landscape and portrait modes successfully, but this is somewhat workaround, real fix is rotating video frames before writing it with muxer, since rotating requires matrix transformations etc. it may be slow on android java (i added TODO into code for future) 👍